tests/test_bidsInterface.py

```diff
@@ -32,25 +39,59 @@ def test_remoteBidsInterface(self):
-        runBidsInterfaceTest(bidsInterface)
-
-
-def runBidsInterfaceTest(bidsInterface):
-    required_metadata = {'subject': '04', 'task': 'story', 'suffix': 'bold', 'datatype': 'func', 'run': 1}
-    other_metadata = {'a1': [1, 'two', 3.0], 
-                      'a2': {'np': numpy.float32(3), 'pyint': 4, 'str': 'five'},
-                      'a3': [6.0, 'seven', numpy.int(8)]}
-    args = (4, 'hello', required_metadata)
-    kwargs = {'mdata': other_metadata, 'test1': 9.0, 'test2': numpy.float32(9), 'test3': 'yes'}
-    res = bidsInterface.testMethod(*args, **kwargs)
-    print(f'testMethod returned {res}')
-    args_py = npToPy(args)
-    kwargs_py = npToPy(kwargs)
-    res[0] == args_py
-    res[1] == kwargs_py
+        dicomStreamTest(bidsInterface)
+        openNeuroStreamTest(bidsInterface)
+
+
+def dicomStreamTest(bidsInterface):
+    # initialize the stream
+    entities = {'subject': '01', 'task': 'test', 'run': 1, 'suffix': 'bold', 'datatype': 'func'}
+    print(f'### {test_sampleProjectDicomPath}')
+    streamId = bidsInterface.initDicomBidsStream(test_sampleProjectDicomPath,
+                                                 "001_000013_{TR:06d}.dcm", 300*1024, **entities)
+
+    for idx in [*range(10), 5, 2, 7]:
+        # get the incremental from the stream
+        streamIncremental = bidsInterface.getIncremental(streamId, volIdx=idx)
+        # read the incremental locally for test comparison
+        dicomPath = os.path.join(test_sampleProjectDicomPath, "001_000013_{TR:06d}.dcm".format(TR=idx))
+        dicomImg = readDicomFromFile(dicomPath)
+        dicomMetadata = getDicomMetadata(dicomImg)
+        dicomMetadata.update(entities)
+        niftiImg = convertDicomImgToNifti(dicomImg)
+        localIncremental = BidsIncremental(niftiImg, dicomMetadata)
+        print(f"Dicom stream check: image {idx}")
+        assert streamIncremental == localIncremental
+    pass
+
+def openNeuroStreamTest(bidsInterface):
+    dsAccessionNumber = 'ds002338'
+    dsSubject = 'xp201'
+    datasetDir = tmpDownloadOpenNeuro(dsAccessionNumber, dsSubject, 1)
+    localEntities = {'subject': dsSubject, 'run': 1, 'suffix': 'bold', 'datatype': 'func'}
+    remoteEntities = {'subject': dsSubject, 'run': 1}
+    localBidsArchive = BidsArchive(datasetDir)
+    streamId = bidsInterface.initOpenNeuroStream(dsAccessionNumber, **remoteEntities)
+    for idx in range(3):
+        streamIncremental = bidsInterface.getIncremental(streamId)
+        localIncremental = localBidsArchive.getIncremental(idx, **localEntities)
+        print(f"OpenNeuro stream check: image {idx}")
+        assert streamIncremental == localIncremental
+
+    for idx in [5, 2, 7]:
+        streamIncremental = bidsInterface.getIncremental(streamId, volIdx=idx)
+        localIncremental = localBidsArchive.getIncremental(idx, **localEntities)
+        print(f"OpenNeuro stream check: image {idx}")
+        assert streamIncremental == localIncremental
+
```

Comment:
Given earlier comment on `-1` behavior, might be good to add a test that goes something like `-1, -1, -1, 8, -1, -1, -1` and ensure you get image `0, 1, 2, 8, 3, 4, 5`, in that order